WootricSDK/WootricSDK/WTRApiClient.m

```diff
@@ -290,10 +290,19 @@ - (void)authenticate:(void (^)(void))authenticated {
         if (accessToken) {
           self->_accessToken = accessToken;
           [self getEndUserWithEmail:^(NSInteger endUserID) {
-            self.userID = endUserID;
+            if (endUserID != 0) {
```

Comment:
You are creating a small forest of `if` here. Are you sure there's no other way around that?

Reply:
```m
  NSURLSessionDataTask *dataTask = [_wootricSession dataTaskWithRequest:urlRequest completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
    if (error) {
      [WTRLogger logError:@"(authentication): %@", error];
      return;
    }

    NSDictionary *responseJSON = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
    [WTRLogger log:@"(authentication): %@", responseJSON];

    if (!responseJSON) {
      authenticated(NO);
      return;
    }

    NSString *accessToken = responseJSON[@"access_token"];

    if (!accessToken) {
      authenticated(NO);
      return;
    }

    self->_accessToken = accessToken;

    [self getEndUserWithEmail:^(NSInteger endUserID) {
      if (endUserID != 0) {
        self.userID = endUserID;
        authenticated(YES);
      } else {
        [WTRLogger logError:@"Error getting/creating End User"];
        authenticated(NO);
      }
    }];
  }];
```
One idea I had was to rewrite this in an early return style. Basically check the error case and return early. This will flatten out the code.